accumulo-iterators/src/test/java/org/vertexium/accumulo/iterator/RowDeduplicationIteratorTest.java

```diff
@@ -0,0 +1,204 @@
+package org.vertexium.accumulo.iterator;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import org.apache.accumulo.core.client.impl.BaseIteratorEnvironment;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorUtil;
+import org.apache.accumulo.core.iterators.SortedMapIterator;
+import org.apache.commons.collections.BufferOverflowException;
+import org.apache.hadoop.io.Text;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.*;
+
+public class RowDeduplicationIteratorTest {
```

Comment:
Can you test interleaved properties.

In time order...
- t1 - prop1: John
- t2 - prop2: Doe
- t3 - prop1: John
- t4 - prop2: Smith

The two `John` values should be collapsed.